### PR TITLE
Tests, Modules, es6, WebSockets, Raven, better UX

### PR DESCRIPTION
**Pull request by juliankrispel**
The story so far:
- installed mocha and jshint and added hello world test
- compilation with babel
- basic websocket stuff

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "rf-chrome-extension",
```
Comment (ukd1) [inline package.json]: is this meant to be named differently than the project?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "version": "0.0.1",
```
Comment (ukd1) [inline package.json]: important?

File: package.json (hunk 1 above)
Comment (juliankrispel) [inline package.json]: it doesn't really matter what the name is I suppose. Since we're not publishing the package on npm

File: package.json (hunk 2 above)
Comment (juliankrispel) [inline package.json]: since we're not publishing on npm no

Comment (juliankrispel) [inline package.json]: but without defining it npm won't let you install anything

**Comment by shosti** [comment]
@juliankrispel @ukd1 mind giving the websocket stuff a quick look?

**Comment by shosti** [comment]
(requires https://github.com/rainforestapp/Rainforest/pull/7231)

File: src/phoenix.js [hunk 3]
```diff
@@ -0,0 +1,730 @@
+// Phoenix Channels JavaScript client
```
Comment (juliankrispel) [inline src/phoenix.js]: we should include the [npm module instead](https://www.npmjs.com/package/phoenix-js)

Comment (shosti) [inline src/phoenix.js]: Unfortunately the module isn't official and seems out of date (there are some API differences that I'm not sure about). I think including the source is safer for now; I'll ask on the mailing list if there's an "official" npm module.

Comment (juliankrispel) [inline src/phoenix.js]: @shosti yea I just realized that when I tried it. I tried including the github repo as an npm module which should work but they fucked up the format of the file so I can't import it :(

**Comment by juliankrispel** [comment]
image

**Pull request merged**